### DISCUSSION TITLE
20230917: "Update addonslist.json" inapprehensible

### DISCUSSION THREAD

**Discussion by klassikmann** [rern/rAudio#1421] [General]
@rern 
```
-	, "version"     : "20230917"
+	, "version"     : "20230828"
```

Really?

**Comment by ceejayemm** [comment]
@rern 

For me the error is:

Errors: Unexpected token ',', "
	, "status""... is not valid JSON 
X
	, "status" : {
		  "hidden"      : [""]
		, "installed"   : ["r1"]
		, "notverified" : [""]
		, "update"      : ["r1"]
		, "online"      : true
	}
}

**Comment by klassikmann** [comment]
@ceejayemm 
I was surprised that in the diff of addonslist.json the new version had been replaced by the old one (instead of vice versa).
EDIT: But now it seems to be fixed.

**Comment by ceejayemm** [comment]
@rern 

I can see that you have released some further updates to the Main code area but because of the error noted above I cannot try these fixes.  Please can you let me know how to fix this issue so that I can move on to see if your latest fixes also sort out some of my other issues.

Thanks

Chris

**Comment by rern** [comment]
A fix for anyone who has updated to 20230917:
`sed -i 's/channel ip/channel data ip/' $dirbash/common.sh`

Chris,
Please try it.

**Reply by ceejayemm** [reply]
@rern 

Thanks for the quick fix. This now allows me to apply the new 20230918 update .

Chris

**Reply by kw-sudo** [reply]
+1 json error. Fix is working!
Oliver

**Comment by klassikmann** [comment]
@ceejayemm @rern 
Confirmed. Update worked after changing common.sh

**Comment by EarsofTin** [comment]
Attempted "sed -i 's/channel ip/channel data ip/' $dirbash/common.sh" fix without success:

```
Errors: Unexpected token ',', "
	, "status""... is not valid JSON CopyX
	, "status" : {
		  "hidden"      : [""]
		, "installed"   : ["r1"]
		, "notverified" : [""]
		, "update"      : ["r1"]
		, "online"      : true
	}
}```

**Reply by cYCL157** [reply]
I have this same issue and was just putting together a post about it -- I did not run the sed command -- I'm sure I can fix the file if I can figure out the location (I was prepping and then figured out it was prepping my mount points too, so my entire nas....).

**Reply by cYCL157** [reply]
the sed command worked as advertised, i just needed to find the file -- which can be found in this folder on my pi -- /srv/http/bash

**Reply by EarsofTin** [reply]
Thanks, once I understood the syntax of "sed" and navigated to the right directory I'm all up to date with 20230918

**Comment by Raitsa** [comment]
Had to reinstall and restore from backup, I could not figure the common.sh fix thingy although I was really close.
All ok now though, thanks.

**Comment by ccaper** [comment]
I am on version '20230620'.  I tried the "sed -i 's/channel ip/channel data ip/' $dirbash/common.sh" fix.  I see common.sh changed, but still getting error when I click the Addons button.  Any advice?

Error:
Errors: No number after minus sign in JSON at position 2270 (line 3 column 25) 
X{ "push" : { "bash" : "" , "url" : "" , "hide" : 1 } , "r1" : { "title" : "rAudio" , "version" : "20230918" , "revision" : [ "Fix - System - I2S Audio setting" , "Fix - Features - Browser on RPi" , "Fix - System - Audiophonics power switch" , "Improve - Switch from NGINX Pushstream to standard WebSocket" ] , "description" : "Updates for rAudio" , "nouninstall" : 1 , "thumbnail" : "/assets/img/coverart.svg" , "sourcecode" : "https://github.com/rern/rAudio" , "installurl" : "https://github.com/rern/rAudio/raw/main/install.sh" , "verify" : "[[ $( cat /srv/http/data/addons/r1 ) < 20230620 ]] && echo 1" , "notverified" : "Please download the latest image file to be able to update." } , "dab" : { "title" : "DAB Radio" , "description" : "Add Digital Audio Broadcasting radio reception capabilities
(RTL-SDR dongle required)" , "thumbnail" : "/assets/img/addons/thumbdab.jpg" , "sourcecode" : "https://github.com/vale-max/rAudio-addons/raw/main/dabradio" , "installurl" : "https://github.com/rern/rAudio-addons/raw/main/dabradio/install.sh" , "hide" : "[[ -e /usr/bin/mediamtx || -e /boot/kernel.img ]] && echo 1" } , "fran" : { "title" : "Webradio - [France Musique](https://www.francemusique.fr/) & [Fip](https://www.fip.fr/)" , "buttonlabel" : [ "plus-circle", "Add" ] , "description" : "Webradio stations by France Musique and Fip" , "thumbnail" : "/assets/img/addons/thumbfran.gif" , "sourcecode" : "https://github.com/rern/rAudio-addons/raw/main/webradio" , "installurl" : "https://github.com/rern/rAudio-addons/raw/main/webradio/install.sh" , "hide" : "[[ -e /srv/http/data/webradio/Fip ]] && echo 1" } , "rank" : { "title" : "Rank Mirror Package Servers" , "description" : "Fix package download errors caused by unreachable servers.
Rank mirror package servers by download speed and latency." , "thumbnail" : "/assets/img/addons/thumbrank.png" , "buttonlabel" : [ "bars", "Rank" ] , "sourcecode" : "https://github.com/rern/rAudio-addons/tree/main/rankmirrors" , "installurl" : "https://github.com/rern/rAudio-addons/raw/main/rankmirrors/rankmirrors.sh" , "option" : { "message" : "Download test for each server:" , "textlabel" : "Seconds" } , "hide" : "[[ -e /boot/kernel.img ]] && echo 1" }
	, "status" : {
		  "hide"        : [[ -e /boot/kernel.img ]] && echo 1]
		, "installed"   : ["r1"]
		, "notverified" : [""]
		, "update"      : ["r1"]
		, "online"      : true
	}
}

**Reply by ceejayemm** [reply]
If you read rern's comment for this fix, its for 20230917:

"A fix for anyone who has updated to 20230917"

Could be why it didn't work for you ?

Chris

**Reply by ccaper** [reply]
Thank you, right after I posted this, I read back over the comments and noticed that.  Any suggestions on a fix if you are still on '20230620'?

**Reply by R101G** [reply]
I could not understand the _sed_ command but, after some trial and error, _cd /srv/http/bash_ followed by _sed -i 's/channel ip/channel data ip/' common.sh_ worked for me.
